mysql-test/suite/rpl/t/rpl_semi_sync_master_shutdown.test

```diff
@@ -42,8 +43,10 @@ source include/rpl_start_server.inc;
 # Clean up
 #
 --connection slave
---source include/stop_slave.inc
---source include/start_slave.inc
+--source include/wait_for_slave_sql_to_start.inc
+--let rpl_allow_error=1
+--source include/wait_for_slave_io_to_start.inc
```

Comment:
With the above change to await the error, I imagine the IO thread will have stopped and that it should be re-started manually, i.e. it should have missed the window for automatic restart.